tests/integration_tests/datasource/api_tests.py

```diff
@@ -0,0 +1,137 @@
+    def get_virtual_dataset(self):
+        return (
+            db.session.query(SqlaTable)
+            .filter(SqlaTable.table_name == "virtual_dataset")
+            .one()
+        )
+
+    @pytest.mark.usefixtures("app_context", "virtual_dataset")
+    def test_get_column_values_ints(self):
+        self.login(username="admin")
+        table = self.get_virtual_dataset()
+        rv = self.client.get(f"api/v1/datasource/table/{table.id}/column/col1/values/")
+        self.assertEqual(rv.status_code, 200)
+        response = json.loads(rv.data.decode("utf-8"))
+        for val in range(10):
+            assert val in response["result"]
+
+    @pytest.mark.usefixtures("app_context", "virtual_dataset")
+    def test_get_column_values_strs(self):
+        self.login(username="admin")
+        table = self.get_virtual_dataset()
+        rv = self.client.get(f"api/v1/datasource/table/{table.id}/column/col2/values/")
+        self.assertEqual(rv.status_code, 200)
+        response = json.loads(rv.data.decode("utf-8"))
+        for val in ["a", "b", "c", "d", "e", "f", "g", "h", "i", "j"]:
+            assert val in response["result"]
+
+    @pytest.mark.usefixtures("app_context", "virtual_dataset")
+    def test_get_column_values_floats(self):
+        self.login(username="admin")
+        table = self.get_virtual_dataset()
+        rv = self.client.get(f"api/v1/datasource/table/{table.id}/column/col3/values/")
+        self.assertEqual(rv.status_code, 200)
+        response = json.loads(rv.data.decode("utf-8"))
+        for val in [1.0, 1.1, 1.2, 1.3, 1.4, 1.5, 1.6, 1.7, 1.8, 1.9]:
+            assert val in response["result"]
+
+    @pytest.mark.usefixtures("app_context", "virtual_dataset")
+    def test_get_column_values_nulls(self):
+        self.login(username="admin")
+        table = self.get_virtual_dataset()
+        rv = self.client.get(f"api/v1/datasource/table/{table.id}/column/col4/values/")
+        self.assertEqual(rv.status_code, 200)
+        response = json.loads(rv.data.decode("utf-8"))
+        self.assertEqual(response["result"], [None])
+
+    @pytest.mark.usefixtures("app_context", "virtual_dataset")
+    def test_get_column_values_invalid_datasource_type(self):
+        self.login(username="admin")
+        table = self.get_virtual_dataset()
+        rv = self.client.get(
+            f"api/v1/datasource/not_table/{table.id}/column/col1/values/"
+        )
+        self.assertEqual(rv.status_code, 400)
+        response = json.loads(rv.data.decode("utf-8"))
+        self.assertEqual(response["message"], "Invalid datasource type: not_table")
+
+    @patch("superset.datasource.api.DatasourceDAO.get_datasource")
+    def test_get_column_values_datasource_type_not_supported(self, get_datasource_mock):
+        get_datasource_mock.side_effect = DatasourceTypeNotSupportedError
+        self.login(username="admin")
+        rv = self.client.get("api/v1/datasource/table/1/column/col1/values/")
```

Comment:
would sending `api/v1/datasource/bogus_type/1/column/col1/values/` result in the same?

Reply:
No, that would raise a `ValueError` on the endpoint, that case is covered by the `test_get_column_values_invalid_datasource_type` above this one.  This exception can be raised if the type is in the `DatasourceType` enum, but not supported by `get_datasource`